TPP.Persistence.MongoDB.Tests/Repos/BadgeRepoTest.cs

```diff
@@ -190,6 +197,33 @@ public async Task FindAllForSaleByCustom_only_returns_badges_for_sale()
             Assert.AreEqual(new List<Badge>() { forSale }, saleList);
         }
 
+        [Test]
+        public async Task can_handle_null_shiny_fields()
+        {
+            IMongoDatabase db = CreateTemporaryDatabase();
+
+            const string id = "590df61373b975210006fcdf";
+            Instant instant = InstantPattern.ExtendedIso.Parse("2017-05-06T16:13:07.314Z").Value;
+
+            BadgeRepo badgeRepo = new BadgeRepo(db, new BadgeLogRepo(db), new MockClock());
+            IMongoCollection<BsonDocument> bsonBadgeCollection = db.GetCollection<BsonDocument>("badges");
+            await bsonBadgeCollection.InsertOneAsync(BsonDocument.Create(new Dictionary<string, object?>
+            {
+                ["_id"] = ObjectId.Parse(id),
+                ["userid"] = "mogi",
+                ["species"] = "1",
+                ["source"] = "manual_creation",
+                ["created_at"] = instant.ToDateTimeUtc(),
+                ["form"] = 0,
+            }));
+
+            IMongoCollection<Badge> badgeCollection = db.GetCollection<Badge>("badges"); ;
+
+            Badge b = await badgeCollection.Find(b => b.Id == id).FirstAsync();
+
+            Assert.AreEqual(false, b.Shiny);
```

Comment:
This is a good test to see that an absent "shiny" field gets properly deserialized as "false" in the model class. Can you add an additional test case that calls `FindAllByCustom` with shiny=false to see whether that filter will also find badges with an unset or null shiny field?